src/ucp/core/ucp_context.c

```diff
@@ -290,13 +295,15 @@ static ucs_status_t ucp_add_tl_resources(ucp_context_h context,
     ucs_status_t status;
     ucp_rsc_index_t i;
 
+    UCP_CONTEXT_THREAD_CS_ENTER(context);
```

Comment:
we need to add this only to **API** functions. no need to add it to internal functions.